File: src/main/java/io/vproxy/vfx/animation/AnimationStateTransferBeginCallback.java
package io.vproxy.vfx.animation;

import io.vproxy.vfx.util.algebradata.AlgebraData;

public interface AnimationStateTransferBeginCallback<T extends AlgebraData<T>> {
    void animationStateTransferBegin(AnimationNode<T> from, AnimationNode<T> to);
}

File: src-ts/src/animation/AnimationStateTransferBeginCallback.ts
// Plain function type to avoid Function.prototype method-name collision.
import type { AlgebraData } from '../util/algebradata/AlgebraData.js';
import type { AnimationNode } from './AnimationNode.js';

export type AnimationStateTransferBeginCallback<T extends AlgebraData<T>> = (
  from: AnimationNode<T>,
  to: AnimationNode<T>,
) => void;
